### ISSUE TITLE
Enable to use git last commit as timestamp

### ISSUE THREAD

**Issue by rmannibucau** [GoogleContainerTools/jib#2317]
The image timestamp is not implicit from the actual project work today requiring some maven/gradle hacks and other plugins. Would be neat to enable jib to use jgit to read last commit date and use that as image timestamp. Would keep tags reproducibles but with a really meaningful date (think about webjars and last modified date http header for example).

**Comment by loosebazooka** [comment]
So you should be able to query the system either at the plugin level or the command line level and inject these parameters in? Let me find a good example.

**Comment by rmannibucau** [comment]
@loosebazooka this is not portable. I can do it with a pre-task (I'm in maven case so another plugin so https://github.com/git-commit-id/git-commit-id-maven-plugin/) but it is not convenient and becomes complex for something trivial to fix in the plugin so think it would be sane to include it, in particular today git became the standard.

**Comment by loosebazooka** [comment]
That's a valid point, but our philosophy has always been to provide the extension points for users to inject whatever values they want without getting specifically into any particular mechanism. I think using the git-commit-id-maven-plugin with the maven-properties-plugin is a valid workaround. However having said that, the design described in https://github.com/GoogleContainerTools/jib/pull/2312 should allow anyone to write a git provider in the future and inject those values into the build plan.

**Comment by rmannibucau** [comment]
Think the solution can tend to create a jib-extension-git-provider module in the project we can add in the buildscript (gradle) or plugin dependencies (maven) which would enable that automatically (through a plain old SPI).
Would keep everyone clean.
But since it is a build plugin it should be provided OOTB (on central even if not a default dependency) IMHO.

**Comment by loosebazooka** [comment]
right, that's kind of how it would work, via published extension jars.

**Comment by ST-DDT** [comment]
AFAICT this is already possible for git and maven:

````xml
<plugin>
	<groupId>pl.project13.maven</groupId>
	<artifactId>git-commit-id-plugin</artifactId>
	<configuration>
		<dateFormat>yyyy-MM-dd'T'HH:mm:ss'Z'</dateFormat>
		<dateFormatTimeZone>UTC</dateFormatTimeZone>
	</configuration>
</plugin>
````
and then:
````xml
<plugin>
	<groupId>com.google.cloud.tools</groupId>
	<artifactId>jib-maven-plugin</artifactId>
	<configuration>
		<container>
			<filesModificationTime>${git.commit.time}</filesModificationTime>
		</container>
	</configuration>
</plugin>
````
IMO this is not too much to configure yourself.

**Comment by rmannibucau** [comment]
@ST-DDT you loose the ability to use that per file with such a config so I think the extension can be way saner, in particular for web resources.

**Comment by ST-DDT** [comment]
I just created an issue related to the timestamp on libraries: https://github.com/GoogleContainerTools/jib/issues/2462

There I proposed a similar solution to [your extension proposal](https://github.com/GoogleContainerTools/jib/issues/2317#issuecomment-596629820):

https://github.com/GoogleContainerTools/jib/issues/2462#issuecomment-628064979
